v8files-extractor.os

```diff
@@ -262,28 +289,46 @@
 	Лог.Отладка("Запускаем распаковку файла");
 	
 	Конфигуратор = Новый УправлениеКонфигуратором();
-	КаталогВременнойИБ = ВременныеФайлы.СоздатьКаталог();
-	Конфигуратор.КаталогСборки(КаталогВременнойИБ);
+	Если КонтекстКонфигуратора = Неопределено Тогда
+		КаталогВременнойИБ = ВременныеФайлы.СоздатьКаталог();
+		Конфигуратор.КаталогСборки(КаталогВременнойИБ);
+	Иначе
+		Конфигуратор.ИспользоватьКонтекст(КонтекстКонфигуратора);
+	КонецЕсли;
+
 	
 	ЛогКонфигуратора = Логирование.ПолучитьЛог("oscript.lib.v8runner");
 	ЛогКонфигуратора.УстановитьУровень(Лог.Уровень());
 	
 	Параметры = Конфигуратор.ПолучитьПараметрыЗапуска();
-	Параметры[0] = "ENTERPRISE";
-	
-	ПутьV8Reader = ОбъединитьПути(ТекущийСценарий().Каталог, "v8Reader", "V8Reader.epf");
-	Лог.Отладка("Путь к V8Reader: " + ПутьV8Reader);
-	Ожидаем.Что(Новый Файл(ПутьV8Reader).Существует()).ЭтоИстина();
-	
-	КоманднаяСтрокаV8Reader = СтрЗаменить("/C""decompile;pathtocf;%1;pathout;%2;convert-mxl2txt;ЗавершитьРаботуПосле;""","%1", Файл.ПолноеИмя);
-	КоманднаяСтрокаV8Reader = СтрЗаменить(КоманднаяСтрокаV8Reader,"%2", ПапкаИсходников.ПолноеИмя);
-	
-	Лог.Отладка("Командная строка V8Reader: " + КоманднаяСтрокаV8Reader);
+	Если ИспользоватьКонфигуратор Тогда
+
+		Параметры[0] = "DESIGNER";
```

Comment:
https://github.com/EvilBeaver/oscript-library/blob/develop/src/v8runner/src/v8runner.os#L513

вот эта штука есть всегда

16 мая 2016 г., 15:19 пользователь Борис <notifications@github.com> написал:

> In v8files-extractor.os
> <https://github.com/xDrivenDevelopment/precommit1c/pull/84#discussion_r63344805>
> :
>
> >  	
> >  	ЛогКонфигуратора = Логирование.ПолучитьЛог("oscript.lib.v8runner");
> >  	ЛогКонфигуратора.УстановитьУровень(Лог.Уровень());
> >  	
> >  	Параметры = Конфигуратор.ПолучитьПараметрыЗапуска();
> > -	Параметры[0] = "ENTERPRISE";
> > -	
> > -	ПутьV8Reader = ОбъединитьПути(ТекущийСценарий().Каталог, "v8Reader", "V8Reader.epf");
> > -	Лог.Отладка("Путь к V8Reader: " + ПутьV8Reader);
> > -	Ожидаем.Что(Новый Файл(ПутьV8Reader).Существует()).ЭтоИстина();
> > -	
> > -	КоманднаяСтрокаV8Reader = СтрЗаменить("/C""decompile;pathtocf;%1;pathout;%2;convert-mxl2txt;ЗавершитьРаботуПосле;""","%1", Файл.ПолноеИмя);
> > -	КоманднаяСтрокаV8Reader = СтрЗаменить(КоманднаяСтрокаV8Reader,"%2", ПапкаИсходников.ПолноеИмя);
> > -	
> > -	Лог.Отладка("Командная строка V8Reader: " + КоманднаяСтрокаV8Reader);
> > +	Если ИспользоватьКонфигуратор Тогда
> > +
> > +		Параметры[0] = "DESIGNER";
>
> В документации это не описано, а в исходники либы на момент написания кода
> я ещё не смотрел. Стоит ли исправлять?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/xDrivenDevelopment/precommit1c/pull/84/files/318c56c1aa384ec512fb8a411634949ac8e9d52a#r63344805>
>